R/finalize_plot.R

```diff
@@ -201,18 +219,23 @@ finalize_plot <- function(input_plot = NULL,
     clip = "on"
   )
 
+  # extract height of legend object within ggplot plot
+  legend_height <- grid::convertUnit(ggpubr::get_legend(processed_plot)$heights[[3]],
+                                     "bigpts",
+                                     valueOnly = TRUE)
+
   # create viewport for plot
   vp.plot <- grid::viewport(
     name = "vp.plot",
-    default.units = "bigpts",
-    # origin shifted over from 0,0 of parent vp per spec
     x = plot_constants$title_width,
     y = plot_constants$margin_v4,
     just = c(0,0),
-    # size is function of remaining size available
+    default.units = "bigpts",
+    height = plot_constants$height - plot_constants$margin_v1_v2 -
+      plot_constants$margin_v4,
     width = plot_constants$width - plot_constants$title_width - plot_constants$margin_h3,
-    height = plot_constants$height - plot_constants$margin_v1_v2 - plot_constants$margin_v4,
-    clip = "on"
+    clip = "on",
+    gp = gpar(col = "black")
```

Comment:
What's the purpose of this gpar function?

Reply:
Oops, that's a holdover from a debugging attempt. Deleted.